src/o365/spo/commands/list/list-sitescript-get.ts

```diff
@@ -0,0 +1,240 @@
+} from '../../../../Command';
+import SpoCommand from '../../SpoCommand';
+import Utils from '../../../../Utils';
+import { Auth } from '../../../../Auth';
+import { ContextInfo, 
+  // ClientSvcResponse, 
+  // ClientSvcResponseContents 
+} from '../../spo';
+import { ListInstance } from './ListInstance';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  listId?: string;
+  listTitle?: string;
+}
+
+class SpoListSitescriptGetCommand extends SpoCommand {
+  public get name(): string {
+    return commands.LIST_SITESCRIPT_GET;
+  }
+
+  public get description(): string {
+    return 'Extracts a site script from a SharePoint list';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.listId = (!(!args.options.listId)).toString();
+    telemetryProps.listTitle = (!(!args.options.listTitle)).toString();
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    const resource: string = Auth.getResourceFromUrl(args.options.webUrl);
+    let formDigest: string;
+
+    if (this.debug) {
+      cmd.log(`Retrieving access token for ${resource}...`);
+    }
+
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((accessToken: string): request.RequestPromise => {
+        if (this.debug) {
+          cmd.log(`Retrieved access token ${accessToken}. Retrieving request digest...`);
+        }
+
+        return this.getRequestDigest(cmd, this.debug);
+      })
+      .then((res: ContextInfo): request.RequestPromise => {
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(res);
+          cmd.log('');
+        }
+
+        formDigest = res.FormDigestValue;
+
+        if (this.verbose) {
+          const list: string = args.options.listId ? encodeURIComponent(args.options.listId as string) : encodeURIComponent(args.options.listTitle as string);
+          cmd.log(`Extracting Site Script from list ${list} in site at ${args.options.webUrl}...`);
+        }
+
+        let requestUrl: string = '';
+
+        if (args.options.listId) {
+          if (this.debug) {
+            cmd.log(`Retrieving List Url from Id '${args.options.listId}'...`);
+          }
+          requestUrl = `${args.options.webUrl}/_api/web/lists(guid'${encodeURIComponent(args.options.listId)}')?$expand=RootFolder`;
+        }
+        else {
+          if (this.debug) {
+            cmd.log(`Retrieving List Url from Title '${args.options.listTitle}'...`);
+          }
+          requestUrl = `${args.options.webUrl}/_api/web/lists/GetByTitle('${encodeURIComponent(args.options.listTitle as string)}')?$expand=RootFolder`;
+        }
+
+        const requestOptions: any = {
+          url: requestUrl,
+          method: 'GET',
```

Comment:
It's not necessary to specify method here, since later we're calling `request.get`